<launch>
  <arg name="main" default="true" />
  <arg name="vive" default="true" />
  <arg name="oculus" default="false" />
  <arg name="head" default="true" />
  <arg name="respeaker" default="true" />
  <arg name="mirror" default="false" />
  <arg name="logging" default="true" />
  <arg name="kinfu" default="false" />
  <arg name="home_dir" default="$(env HOME)" />

  <!-- vive -->
  <arg name="b_lighthouse_id" />
  <arg name="c_lighthouse_id" />
  <arg name="l_controller_id" />
  <arg name="r_controller_id" />

  <!-- workspace -->
  <arg name="workspace_max_x" default="false" />
  <arg name="workspace_min_x" default="false" />
  <arg name="workspace_max_y" default="false" />
  <arg name="workspace_min_y" default="false" />
  <arg name="workspace_max_z" default="false" />
  <arg name="workspace_min_z" default="false" />

  <!-- camera-->
  <arg name="CAMERA_INFO" default="/head_mount_kinect/qhd/camera_info" />
  <arg name="RGB_CAMERA_INFO" default="/head_mount_kinect/qhd/camera_info" />
  <arg name="IMAGE_COLOR" default="/head_mount_kinect/qhd/image_color_rect" />
  <arg name="IMAGE_DEPTH" default="/head_mount_kinect/qhd/image_depth_rect" />
  <arg name="image_left" default="/virtual_camera/left/rviz_image" />
  <arg name="image_right" default="/virtual_camera/right/rviz_image" />
  <arg name="DECOMP_CAMERA_INFO"
       default="/kinect_head_decompressed/depth_registered/camera_info" />
  <arg name="DECOMP_RGB_CAMERA_INFO"
       default="/kinect_head_decompressed/rgb/camera_info" />
  <arg name="DECOMP_IMAGE_DEPTH"
       default="/kinect_head_decompressed/depth_registered/image_rect" />
  <arg name="DECOMP_IMAGE_COLOR" default="/kinect_head_decompressed/rgb/image_rect_color" />
  <arg name="DECOMP_POINTS" default="/kinect_head_decompressed/depth_registered/points" />
  <!-- prosilica -->
  <!--
  <arg name="PROSILICA_RGB_CAMERA_INFO"
       default="/prosilica/camera_info" />
  <arg name="PROSILICA_IMAGE_COLOR"
       default="/prosilica/image_rect" />
  <arg name="PROSILICA_DECOMP_RGB_CAMERA_INFO"
       default="/prosilica_decompressed/rgb/camera_info" />
  <arg name="PROSILICA_DECOMP_IMAGE_COLOR"
       default="/prosilica_decompressed/rgb/image_rect_color" />
  -->

  <group ns="kinect_head_decompressed">
    <node name="camera_info_relay" pkg="topic_tools" type="relay"
          args="$(arg CAMERA_INFO) $(arg DECOMP_CAMERA_INFO)" />
    <node name="rgb_camera_info_relay" pkg="topic_tools" type="relay"
          args="$(arg RGB_CAMERA_INFO) $(arg DECOMP_RGB_CAMERA_INFO)" />
    <node name="depth_image_transport" pkg="image_transport" type="republish"
          args="compressed in:=$(arg IMAGE_DEPTH) raw out:=$(arg DECOMP_IMAGE_DEPTH)" />
    <node name="rgb_image_transport" pkg="image_transport" type="republish"
          args="compressed in:=$(arg IMAGE_COLOR) raw out:=$(arg DECOMP_IMAGE_COLOR)" />
    <node name="point_cloud_xyzrgb" pkg="nodelet" type="nodelet"
          args="standalone depth_image_proc/point_cloud_xyzrgb" output="screen">
      <param name="queue_size" value="100" />
    </node>
  </group>

  <include file="$(find eus_vive)/launch/pr2/pr2_vr.launch" pass_all_args="true" />

  <node name="pr2_vr_rosbag_record" pkg="rosbag" type="record"
        args="/eus_vive/status /tf /joint_states
              /head_mount_kinect/qhd/image_color_rect/compressed
              /head_mount_kinect/qhd/camera_info
              /head_mount_kinect/qhd/image_depth_rect/compressed
              --output-prefix $(arg home_dir)/.ros/eus_vive/rosbag/pr2_vr"
        output="screen" if="$(arg logging)" />

</launch>
